packages/graphql-testing/src/MockApolloLink.ts

```diff
@@ -0,0 +1,65 @@
+import {ApolloLink, Observable, Operation} from 'apollo-link';
+import {ExecutionResult, GraphQLError} from 'graphql';
+import {GraphQLMock, MockGraphQLFunction} from './types';
+
+export default class MockApolloLink extends ApolloLink {
+  constructor(private mock: GraphQLMock) {
+    super();
+  }
+
+  request(operation: Operation) {
+    return new Observable(obs => {
+      const {mock} = this;
+      const {operationName = ''} = operation;
+
+      let response: object | null = null;
+
+      if (typeof mock === 'function') {
+        response = mock(operation);
+      } else {
+        const mockForOperation = mock[operationName || ''];
+
+        response =
+          typeof mockForOperation === 'function'
+            ? (mockForOperation as MockGraphQLFunction)(operation)
```

Comment:
Do we need the cast?

Reply:
Unfortunately yes...right now its `Function | MockGraphQLFunction`